CNV-80339: DataVolume docs (#117496)
openshift/openshift-docs: modules/ref-datavolume-source-fields.adoc

// Module included in the following assemblies:
//
// * virt/creating_vm/virt-import-dv-vms.adoc

:_mod-docs-content-type: REFERENCE
[id="ref-datavolume-source-fields_{context}"]
= DataVolume source fields

[role="_abstract"]
You can configure a `DataVolume` object to import a virtual machine (VM) image from several source types.
Each source type has its own set of fields that control where and how CDI retrieves the image.

> == Registry source fields

Author: ```suggestion
[id="ref-datavolume-registry-fields_{context}"]
== Registry source fields
```

Author: Done.

The `source.registry` source type imports a VM image from a container registry.

.`source.registry` fields
[cols="1,1,3",options="header"]
|===
|Field |Type |Description

|`url`
|String
|Container registry URL of the image, specified with the `docker://` prefix. For example, `docker://registry.example.com/my-image:latest`.

|`pullMethod`
|String
|Method for pulling the image. Specify `pod` to use the CDI importer pod or `node` to use the node's container image cache. The default value is `pod`.

|`secretRef`
|String
|Name of a `Secret` object that stores the credentials for the container registry.

|`certConfigMap`
|String
|Name of a `ConfigMap` object that stores custom certificate authority (CA) certificates for the container registry.

|`platform.architecture`
|String
|Target CPU architecture when importing from a multi-architecture image index. Supported values are `amd64`, `arm64`, and `s390x`. This field is a Technology Preview feature.
|===

> == HTTP source fields

Author: ```suggestion
[id="ref-datavolume-http-fields_{context}"]
== HTTP source fields
```

Author: Done.

The `source.http` source type imports a VM image from an HTTP or HTTPS URL.

.`source.http` fields
[cols="1,1,3",options="header"]
|===
|Field |Type |Description

|`url`
|String
|HTTP or HTTPS URL of the VM image.

|`secretRef`
|String
|Name of a `Secret` object that stores Basic authentication credentials for the HTTP server.

|`certConfigMap`
|String
|Name of a `ConfigMap` object that stores custom certificate authority (CA) certificates for the HTTP server.

|`extraHeaders`
|List of strings
|Additional HTTP headers to include in the image download request.

|`secretExtraHeaders`
|List of strings
|Names of `Secret` objects that store additional HTTP headers to include in the image download request.
|===

> == S3 source fields

Author: ```suggestion
[id="ref-datavolume-s3-fields_{context}"]
== S3 source fields
```

Author: Done.

The `source.s3` source type imports a VM image from an S3-compatible object storage bucket.

.`source.s3` fields
[cols="1,1,3",options="header"]
|===
|Field |Type |Description

|`url`
|String
|S3 URL of the VM image.

|`secretRef`
|String
|Name of a `Secret` object that stores the S3 access key ID and secret access key.

|`certConfigMap`
|String
|Name of a `ConfigMap` object that stores custom certificate authority (CA) certificates for the S3 endpoint.
|===

> == Blank source

Author: ```suggestion
[id="ref-datavolume-blank-fields_{context}"]
== Blank source
```

Author: Done.

The `source.blank` source type creates an empty disk with no data.
This source type has no configurable fields.

> == PVC source fields

Author: ```suggestion
[id="ref-datavolume-pvc-fields_{context}"]
== PVC source fields
```

Author: Done.

The `source.pvc` source type clones the contents of an existing persistent volume claim (PVC).

.`source.pvc` fields
[cols="1,1,3",options="header"]
|===
|Field |Type |Description

|`namespace`
|String
|Namespace of the source PVC.

|`name`
|String
|Name of the source PVC.
|===

> == VolumeSnapshot source fields

Author: ```suggestion
[id="ref-datavolume-volumesnapshot-fields_{context}"]
== VolumeSnapshot source fields
```

Author: Done.

The `source.snapshot` source type creates a volume from an existing `VolumeSnapshot` object.

.`source.snapshot` fields
[cols="1,1,3",options="header"]
|===
|Field |Type |Description

|`namespace`
|String
|Namespace of the source `VolumeSnapshot` object.

|`name`
|String
|Name of the source `VolumeSnapshot` object.
|===

> == Supported content types

Author: ```suggestion
[id="ref-datavolume-supported-content-types_{context}"]
== Supported content types
```

Author: Done.

// Module included in the following assemblies:
//
// * virt/creating_vm/virt-import-dv-vms.adoc

:_mod-docs-content-type: REFERENCE
[id="ref-datavolume-source-fields_{context}"]
= `DataVolume` source fields

[role="_abstract"]
You can configure a `DataVolume` object to import a virtual machine (VM) image from several source types.

> Each source type has its own set of fields that control where and how CDI retrieves the image.

Reviewer: "CDI" is used without expansion on its first occurrence in this module. Each module is consumed independently, so acronyms must be expanded on first use within the module.
```suggestion
Each source type has its own set of fields that control where and how the Containerized Data Importer (CDI) retrieves the image.
```

.`source.imageio` fields
[cols="1,1,3",options="header"]
|===
|Field |Type |Description

|`url`
|String
|URL of the oVirt Engine API endpoint. For example, `\https://ovirt-engine.example.com/ovirt-engine/api`.

|`diskId`
|String
|ID of the oVirt disk to import.

|`secretRef`
|String
|Name of a `Secret` object that stores the credentials to access the oVirt Engine.

|`certConfigMap`
|String
|Name of a `ConfigMap` object that stores CA certificates for the oVirt Engine.
|===

[id="ref-datavolume-vddk-fields_{context}"]
== VDDK source fields

The `source.vddk` source type imports a disk image from a VMware vSphere environment by using the Virtual Disk Development Kit (VDDK).

.`source.vddk` fields
[cols="1,1,3",options="header"]
|===
|Field |Type |Description

|`url`
|String
|URL of the vCenter or ESXi host.

|`uuid`
|String
|UUID of the virtual machine in vCenter or ESXi.

|`backingFile`
|String
|Path to the virtual hard drive to import from vCenter or ESXi.

|`thumbprint`
|String
|TLS certificate thumbprint of the vCenter or ESXi host.

|`secretRef`
|String
|Name of a `Secret` object that stores the username and password for the vCenter or ESXi host.

|`initImageURL`
|String

> |URL of a container image with an extracted VDDK library. Overrides the `v2v-vmware` config map.

Reviewer: "config map" (lowercase, two words) is inconsistent with the Kubernetes resource name `ConfigMap`, which is used everywhere else in this PR and in the surrounding table cells. Kubernetes resource names should match their API spelling.
```suggestion
|URL of a container image with an extracted VDDK library. Overrides the `v2v-vmware` `ConfigMap`.
```

> |URL of a container image with an extracted VDDK library. Overrides the `v2v-vmware` `ConfigMap`.

Author: ```suggestion
|URL of a container image with an extracted VDDK library. Overrides the `v2v-vmware` config map.
```
exception per https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting
